microdnf/commands/group/group_info.cpp

```diff
@@ -63,9 +63,10 @@ void GroupInfoCommand::run() {
 
     // Filter by patterns if given
     if (group_specs_str.size() > 0) {
-        auto query_names = libdnf::comps::GroupQuery(query);
+        libdnf::comps::GroupQuery query_names(query);
```

Comment:
We can use two statements
` auto upgradable_package_query = package_query;`
or
`libdnf::comps::GroupQuery upgradable_package_query(query);`
Which do you thing is more transparent?

Reply:
C++ allows many ways to write something. New versions of C++03, ..., C++17, ... are being created and the rules are evolving.
Imagine a situation where constructors and assign operators have a side effect. Variable `b` is not of type `A`. And now a few examples:
```
A a{b};  // Parametrized constructor is used - 1 operation
A a(b);  // Parametrized constructor is used - 1 operation

A a; a = b;  // Default constructor is used;
             // the parametrized constructor if needed (not suitable copy assignment operator is available);
             // then the copy assignment operator - 2 or 3 operations

A a; a = A(b);  // Default constructor is used;
                // the parametrized constructor is used;
                // then the copy assignment operator - 3 operations
```
And now think about the following notations:
```
A a = b;
A a = A(b);
A a = A(A(A(A(b))));
auto a = b;
auto a = A(b);
auto a = A(A(A(A(b))));
```
I know the results.
However, **I prefer to use `A a{b};` or `A a(b);`**. These notations are short and more precise about what will be done.

Reply:
And the difference between `()` and `{}` is another story.
In some cases, the meaning is the same. Sometimes only one of these notation can be used. And sometimes both, but each notation will give a different result.